src/main/java/gomedic/logic/parser/findcommandparser/FindActivityCommandParser.java

```diff
@@ -0,0 +1,59 @@
+package gomedic.logic.parser.findcommandparser;
+
+import gomedic.commons.core.Messages;
+import gomedic.logic.commands.findcommand.FindActivityCommand;
+import gomedic.logic.commands.findcommand.FindDoctorCommand;
+import gomedic.logic.parser.Parser;
+import gomedic.logic.parser.exceptions.ParseException;
+import gomedic.model.util.ActivityTitleContainsKeywordsPredicate;
+
+
+import java.util.Arrays;
+
+import static gomedic.logic.parser.CliSyntax.*;
+
+/**
+ * Parses input arguments and creates a new FindActivityCommand object
+ */
+public class FindActivityCommandParser implements Parser<FindActivityCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FindCommand
+     * and returns a FindCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FindActivityCommand parse(String args) throws ParseException {
```

Comment:
Maybe the `EditCommandParser` classes' implementations could be of help here, since they also deal with optional flags that the users may or may not include in their input, as well as flags that are inputted in any arbitrary order. 
(E.g. `edit i/D001 n/changed name p/1111111 de/Xray` and `edit i/D001 de/Xray p/1111111  n/changed name` both work)

More specifically, the use of `ArgumentMultimap` and `ArgumentTokenizer` could help abstract away the implementation details of parsing the arguments.